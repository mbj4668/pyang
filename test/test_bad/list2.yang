module list2 {
  yang-version 1.1;
  namespace "urn:test:list2";
  prefix "l";

  choice test {
    default node;
    case node {
      leaf  mandatory {
        type string;
        mandatory true; // error: no mandatory under default case
      }
      list mandatory2 { // error: no mandatory under default case
        min-elements "3";
        key "name";
        leaf name {
          type string;
        }
      }
      anyxml aa { // error: no mandatory under default case
        mandatory true;
      }
      anydata bb { // error: no mandatory under default case
        mandatory true;
      }
      container x {
        anyxml cc { // error: no mandatory under default case
          mandatory true;
        }
        anydata dd { // error: no mandatory under default case
          mandatory true;
        }
        container y {
          leaf-list m {
            min-elements "1"; // error: no mandatory under default case
            type string;
          }
        }
      }
    }
  }
  
} 
