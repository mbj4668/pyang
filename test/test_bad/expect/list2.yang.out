list2.yang:9: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:13: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:20: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:23: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:27: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:30: error: MANDATORY_NODE_IN_DEFAULT_CASE
list2.yang:34: error: MANDATORY_NODE_IN_DEFAULT_CASE
